mahjong/record/utils/event.py

```diff
@@ -112,3 +113,99 @@ def __str__(self):
+                return [{'event_type': 'PENG ', 'player': player,
+                         'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                         'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                             [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+            elif type_of.chakan:
+                return [{'event_type': 'GANG', 'player': player,
+                         'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                         'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                             [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+            elif type_of.nuki:
+                return [{'event_type': 'KITA ', 'player': player,
+                         'player_show': str([tile_from_tenhou(i) for i in self_tile][0]),
+                         'player_open': str([tile_from_tenhou(i) for i in self_tile][0])}]
+            else:
+                if len(borrow_tile) == 0:
+                    return [{'event_type': 'ANGAN', 'player': player,
+                             'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                             'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                             'origin': source}]
+                else:
+                    return [{'event_type': 'MNGAN', 'player': player,
+                             'player_show': ''.join([str(tile_from_tenhou(int(i))) for i in self_tile]),
+                             'player_open': ''.join([str(tile_from_tenhou(int(i))) for i in borrow_tile]) + ''.join(
+                                 [str(tile_from_tenhou(int(i))) for i in self_tile]), 'origin': source}]
+
+        if is_richii(event):
+            player = attrs['who']
+            status = attrs['step']
+            if status == '1':
+                return [{'event_type': 'CALRC', 'player': player}]
+            if status == '2':
+                richi = {'event_type': 'RICHI', 'player': player}
+                score = {'event_type': 'SCORE', 'player': player, 'score': '-1000'}
+                return [richi, score]
+
+        if is_dora_indicator_event(event):
+            tile = tile_from_tenhou(int(attrs['hai']))
+            return [{'event_type': 'DORA ', 'player': '0', 'player_see': tile, 'player_show': tile}]
+
+        if is_somebody_win_game(event):
+            player = attrs['who']
+            loser = attrs['fromWho']
+            score = number_list(event.attrib['ten'])[1]
+            return [{'event_type': 'WIN  ', 'player': player, 'origin': loser}]
+
+        if is_nobody_win_game(event):
+            return [{'event_type': 'EVEN ', 'origin': DRAWN_TYPES[attrs['type']] if 'type' in attrs else ''}]
+
+        return ['UNPARSED']
```

Comment:
`'UNPARSED'` is a plain string, which may cause issues when convert to `pandas.DataFrame`. consider return `[]`, and perform a `logging.warn()` to mention something (which?) not recognized.